clients/drcachesim/tracer/raw2trace.h

```diff
@@ -1066,12 +1066,26 @@ template <typename T> class trace_converter_t {
                             int_modoffs, cur_modoffs);
                 // Because we increment the instr fetch first, the signal modoffs may be
                 // less than the current for a memref fault.
-                if (int_modoffs == cur_modoffs ||
+                // It might also be 0, which means it should be on the first instruction.
+                if (int_modoffs == 0 || int_modoffs == cur_modoffs ||
                     int_modoffs + instr_length == cur_modoffs) {
                     impl()->log(4, "Signal/exception interrupted the bb @ +" PIFX "\n",
                                 int_modoffs);
                     append = true;
                     *interrupted = true;
+                    if (int_modoffs == 0) {
```

Comment:
A callback will always be in between bbs: it won't get here.  I thought about this in my head and apparently never put it into a comment.  I don't have a good way to add an assert or check though since at this point we've lost what kind of kernel event it was, and in the tracer we can't easily tell whether we're done with the prior bb entries in the buffer without doing a lot of decoding and sleuthing.

I'll augment the comment.  I think that's all we can reasonably do.